### ISSUE TITLE
Treat _ and * as brackets

### ISSUE THREAD

**Issue by reknih** [typst/typst#1145]
### Description

The web app should always insert pairs of `_` and `*` and support adding them around the current selection.

### Use Case

Editing faster.

**Comment by jassler** [comment]
If I can chime in, I feel that the editor is sometimes a little too generous with generating these pairs. As a small example, within strings I usually don't want or expect the `_`, `*`, `$`, `"`, etc., characters to be duplicated. Duplicate `*` could become irritating in regexes.

**Comment by laurmaedje** [comment]
True, the editor is not aware of the context when doing it. That's also the reason why `*` doesn't do produce two stars at the moment: It would be very strange in code. We first need to build the infrastructure for this to be context-sensitive.
